### DISCUSSION TITLE
Rare Allele Discovery

### DISCUSSION THREAD

**Discussion by sidevshiy** [maize-genetics/phg_v2#192] [Q&A] [answered]
Hello,

I have a couple of questions about the functionality available in PHG. Specifically, I am interested in whether there is a way to extract imputed VCFs that contain SNP variances. Could you please provide guidance on how to achieve this if it is possible?

In your architecture diagram, there is a section labeled "Rare Allele Discovery," which is also mentioned in the description of the `phg create-fasta-from-hvcf` command. As I understand it, this process (including aligning and using GATK) needs to be performed manually. Is my understanding correct? Additionally, are there any plans to integrate this process into the PHG workflow?

**Comment by lynnjo** [comment, marked as answer]
The Rare Allele Discovery pipeline is still under development.  We are determining the best ways to approach this and do not yet have commands for it.

In terms of imputed VCFs:  There is a command named "hvcf2gvcf" that takes the hvcf file created from the imputed path and turns it into a g.vcf.  This should give you the variants you seek.

**Reply by sidevshiy** [reply]
Thank you for your response!

I have a question about the reference allele in these gvcf files. I assumed it was the reference allele from my reference genome, but when I ran bcftools norm --check-ref, I encountered many mismatches with my reference. Could you clarify what the reference allele in gvcf files there?

**Reply by lynnjo** [reply]
Are you seeing issues with the reference allele in the gvcf files created via the create-maf-vcf command, or only the reference allele  in the gvcf file created from hvcf2gvcf?

**Reply by sidevshiy** [reply]
hvcf2gvcf command. The interesting thing is that in hVCF files after alignments everyting is okay, but positions in gVCF files after hvcf2gvcf are not all the same as in reference.

bcftools norm --check-ref w --fasta-ref /path/to/Ref.fa path/to/gvcf/sample.g.vcf -o test.vcf
REF_MISMATCH	Chr1A	1193710	T	A
REF_MISMATCH	Chr1A	1202566	A	C
REF_MISMATCH	Chr1A	1228692	G	A
REF_MISMATCH	Chr1A	1241315	G	T
REF_MISMATCH	Chr1A	1329686	C	A
REF_MISMATCH	Chr1A	1351172	A	T
REF_MISMATCH	Chr1A	1565115	T	C
REF_MISMATCH	Chr1A	1576373	T	A
REF_MISMATCH	Chr1A	1933410	G	T
REF_MISMATCH	Chr1A	1943709	G	C
REF_MISMATCH	Chr1A	1952918	G	C
REF_MISMATCH	Chr1A	1958897	A	C
REF_MISMATCH	Chr1A	2053764	T	A
REF_MISMATCH	Chr1A	2065238	A	C

**Reply by lynnjo** [reply]
The gvcf file created in the hvcf2gvcf command pulls the data from the original gvcf files previously stored in the db.  Can you verify the existing (original) gvcf files have the correct value at those positions?

**Reply by sidevshiy** [reply]
Yes, when I run this command (bcftools norm --check-ref w --fasta-ref /path/to/Ref.fa path/to/gvcf/sample.g.vcf -o test.vcf) with gvcfs (actually the extentions is vcf when it pull those anchorwave alignmets from db) it doesn't give me a warnings and positions are correct. hvcf files also have correct positions. I believe that there is an error in hvcf2gvcf command.

**Reply by lynnjo** [reply]
ok - I will take a look at this.

**Reply by lynnjo** [reply]
hi @sidevshiy 
I have found and committed a fix for the wrong ref allele problem.   The fix is in the current version of phgv2 (2.3.15.152).  WIll you check this out and verify it works for you?  Thanks for all your feedback.

**Reply by sidevshiy** [reply]
@lynnjo Hello, sorry for long time with no answer. Yes, it works now. Thank you!
